ks-svarut/src/main/java/no/difi/meldingsutveksling/ks/mapping/ForsendelseMapper.java

```diff
@@ -67,19 +69,30 @@ private NoarkMetadataFraAvleverendeSakssystem metaDataFrom(MeldingType meldingTy
         metadata.withJournalpostnummer(Integer.valueOf(meldingType.getJournpost().getJpJpostnr()));
         metadata.withJournalposttype(meldingType.getJournpost().getJpNdoktype());
         metadata.withJournalstatus(meldingType.getJournpost().getJpStatus());
-        metadata.withJournaldato(journalDatoFrom(meldingType.getJournpost().getJpDokdato()));
+        metadata.withJournaldato(journalDatoFrom(meldingType.getJournpost().getJpJdato()));
+        metadata.withDokumentetsDato(journalDatoFrom(meldingType.getJournpost().getJpDokdato()));
+        metadata.withTittel(meldingType.getJournpost().getJpOffinnhold());
+
+        Optional<AvsmotType> avsender = getAvsender(meldingType);
+        avsender.map(a -> a.getAmNavn()).ifPresent(metadata::withSaksbehandler);
 
         return metadata.build();
     }
 
-    private XMLGregorianCalendar journalDatoFrom(String jpDokdato) {
-        LocalDateTime localDateTime = LocalDateTime.of(LocalDate.parse(jpDokdato), LocalTime.of(0, 0));
+
+    private Optional<AvsmotType> getAvsender(MeldingType meldingType) {
+        List<AvsmotType> avsmotlist = meldingType.getJournpost().getAvsmot();
+        return avsmotlist.stream().filter(f -> f.getAmIhtype().equals("0")).findFirst();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "0" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1132)